pkg/volume/gce_pd/gce_util.go

```diff
@@ -123,45 +123,42 @@ func (util *GCEDiskUtil) DetachDisk(c *gcePersistentDiskCleaner) error {
 // Attaches the specified persistent disk device to node, verifies that it is attached, and retries if it fails.
 func attachDiskAndVerify(b *gcePersistentDiskBuilder, sdBeforeSet sets.String) (string, error) {
 	devicePaths := getDiskByIdPaths(b.gcePersistentDisk)
-	var gce cloudprovider.Interface
+	var gceCloud *gce_cloud.GCECloud
 	for numRetries := 0; numRetries < maxRetries; numRetries++ {
-		if gce == nil {
-			var err error
-			gce, err = cloudprovider.GetCloudProvider("gce", nil)
-			if err != nil || gce == nil {
+		if gceCloud == nil {
+			gceCloudProvider, err := cloudprovider.GetCloudProvider("gce", nil)
+			if err != nil || gceCloudProvider == nil {
 				// Retry on error. See issue #11321
 				glog.Errorf("Error getting GCECloudProvider while attaching PD %q: %v", b.pdName, err)
-				gce = nil
 				time.Sleep(errorSleepDuration)
 				continue
 			}
+
+			// The conversion must be safe otherwise bug in GetCloudProvider()
+			gceCloud = gceCloudProvider.(*gce_cloud.GCECloud)
 		}
 
 		if numRetries > 0 {
 			glog.Warningf("Timed out waiting for GCE PD %q to attach. Retrying attach.", b.pdName)
 		}
 
-		if err := gce.(*gce_cloud.GCECloud).AttachDisk(b.pdName, b.readOnly); err != nil {
+		if err := gceCloud.AttachDisk(b.pdName, b.readOnly); err != nil {
 			// Retry on error. See issue #11321. Continue and verify if disk is attached, because a
 			// previous attach operation may still succeed.
 			glog.Errorf("Error attaching PD %q: %v", b.pdName, err)
```

Comment:
Is there any point in verifying the device path if we know confidently that the disk wasn't attached? Unless I'm misunderstanding the code, we should just `continue` in this case.